migration/identity-migration/src/main/java/io/camunda/migration/identity/handler/sm/GroupMigrationHandler.java

```diff
@@ -0,0 +1,229 @@
+        assignedUserCount.get(),
+        totalUserAssignmentAttempts.get(),
+        assignedRoleCount.get(),
+        totalRoleAssignmentAttempts.get());
+  }
+
+  private void assignUsersToGroup(final String groupId, final String targetGroupId) {
+    final var users = managementIdentityClient.fetchGroupUsers(groupId);
+    totalUserAssignmentAttempts.addAndGet(users.size());
+
+    users.forEach(
+        user -> {
+          try {
+            final var username = user.getUsername();
+            logger.debug(
+                "Assigning User: {} with username: {} to Group: {}",
+                user.getId(),
+                username,
+                targetGroupId);
+            final var groupMember = new GroupMemberDTO(targetGroupId, username, EntityType.USER);
+            groupServices.assignMember(groupMember).join();
+            assignedUserCount.incrementAndGet();
+          } catch (final Exception e) {
+            if (!isConflictError(e)) {
+              throw new MigrationException(
+                  String.format(
+                      "Failed to assign user with ID '%s' to group with ID '%s'",
+                      user.getUsername(), targetGroupId),
+                  e);
+            }
+            logger.debug(
+                "User with ID '{}' already assigned to group '{}', skipping assignment.",
+                user.getUsername(),
+                targetGroupId);
+          }
+        });
+  }
+
+  private void assignRolesToGroup(final String groupId, final String targetGroupId) {
+    final var roles = managementIdentityClient.fetchGroupRoles(groupId);
+    totalRoleAssignmentAttempts.addAndGet(roles.size());
+
+    roles.forEach(
+        role -> {
+          try {
+            final var normalizedRoleId = normalizeID(role.name());
+            logger.debug("Assigning Role: {} to Group: {}", normalizedRoleId, targetGroupId);
+            final var groupMember =
+                new GroupMemberDTO(targetGroupId, normalizedRoleId, EntityType.ROLE);
+            groupServices.assignMember(groupMember).join();
+            assignedRoleCount.incrementAndGet();
+          } catch (final Exception e) {
+            if (!isConflictError(e)) {
+              throw new MigrationException(
+                  String.format(
+                      "Failed to assign role '%s' to group with ID '%s'",
+                      role.name(), targetGroupId),
+                  e);
+            }
+            logger.debug(
+                "Role '{}' already assigned to group '{}', skipping assignment.",
+                role.name(),
+                targetGroupId);
+          }
+        });
+  }
+
+  private void createAuthorizationsForGroup(final String groupId, final String targetGroupId) {
+    final List<Authorization> authorizations;
+    try {
+      authorizations = managementIdentityClient.fetchGroupAuthorizations(groupId);
+    } catch (final Exception e) {
+      if (!isNotImplementedError(e)) {
```

Comment:
❓ that's an exception thrown from the client if the method is not implemented? Or is this actually thrown when the endpoint is not available on server side? I'm wondering as the type is from apache commons and I don't see how the client would convert a http error to that?

Reply:
There is this interceptor `EndpointUnavailableInterceptor` that throws this error. I thenk they already faced the same problem 😄